app/views/users/show.html.erb

```diff
@@ -1,16 +1,17 @@
 <section class="user-section">
   <h2><%= "Name: #{@user.name}" %>
-    <% if @friend %>
+    <% if @friend && !(@user.id.equal?current_user.id) %>
```

Comment:
- [ ] Good job showing the button only when the user is not the current user. But let's organize your code a bit.
View logic should never be in the views, it should be in helper methods. Make sure all those if/else statements are inside helper methods.